pkg/kubelet/certificate/bootstrap/bootstrap.go

```diff
@@ -86,9 +92,16 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
 	if err != nil {
 		return err
 	}
-	if err := certutil.WriteCert(certPath, certData); err != nil {
+	if err = certutil.WriteCert(certPath, certData); err != nil {
+		return err
+	}
+	cert, err := x509.ParseCertificate(certData)
+	if err != nil {
 		return err
 	}
+	if cert.NotAfter.Sub(time.Now()) <= time.Duration(0) {
```

Comment:
as commented in https://github.com/kubernetes/kubernetes/pull/50298, client side validation of cert expiry and refusal to create the client isn't how we want to handle this. there is client cert rotation that can request a new cert in this condition, but that is not done here